scripts/generateProptypes.ts

```diff
@@ -135,6 +135,7 @@ const useExternalDocumentation: Record<string, '*' | string[]> = {
   ],
   Tab: ['disableRipple'],
   ToggleButton: ['disableRipple'],
+  Typography: ['className', 'component'],
```

Comment:
For some reason, these two props were not generated by the `yarn proptypes`. I noticed that if I add default values, they are being added to the `propTypes`. Not sure if this is the best fix :\

Reply:
cc @eps1lon

Reply:
Looking at it.

Reply:
Fixed in aedf93f00ee32e85f41f354fac36b95334888576

Before we had an easy time via static analysis to determine what the input props are. Now we actually need to employ flow analysis which is somewhat harder so I short-circuited by considering `inProps` as a magic name (https://github.com/mui-org/material-ui/pull/23571). The name was used based on precedent (Modal + SwipeableDrawer). It works because we are very accurate with the patterns for component declarations (no spread in function arguments, props are named `props` etc). 

However, this PR used `inputProps` instead and we don't have any explicit warnings about this other than: .propTypes changes. I don't believe it's worth adding documentation for typescript-to-proptypes considering how documentation for internal tools is usually ignored. Let's see if this comes back again at which point we can think about some safe-guards.


Somewhat predicated this issue though: https://github.com/mui-org/material-ui/pull/23571#discussion_r524549162

Reply:
I believe if the name is an issue we should update the other components to `inProps` too.